app/pages/home-for-user/index.jsx

```diff
@@ -0,0 +1,242 @@
+import React from 'react';
+import BlurredImage from './blurred-image';
+import Pullout from 'react-pullout';
+import getUserRibbonData from '../../lib/get-user-ribbon-data';
+import CircleRibbon from './circle-ribbon';
+import RecentProjectsSection from './recent-projects';
+import RecentCollectionsSection from './recent-collections';
+import RecentMessagesSection from './recent-messages';
+import MyBuildsSection from './my-builds';
+import ProjectStats from './project-stats';
+import qs from 'qs';
+import HomePageSocial from '../home-not-logged-in/social';
+import NewsSection from './news-pullout';
+
+import style from './index.styl';
+void style;
+
+const SECTIONS = {
+  projects: RecentProjectsSection,
+  collections: RecentCollectionsSection,
+  messages: RecentMessagesSection,
+  builds: MyBuildsSection,
+};
+
+const HomePageForUser = React.createClass({
+  propTypes: {
+    user: React.PropTypes.object,
+    location: React.PropTypes.object,
+  },
+
+  contextTypes: {
+    setAppHeaderVariant: React.PropTypes.func,
+  },
+
+  getDefaultProps() {
+    return {
+      user: {},
+    };
+  },
+
+  getInitialState() {
+    return {
+      backgroundSrc: '',
+      avatarSrc: '',
+      showNews: false,
+      ribbonData: [],
+      loading: false,
+      error: null,
+      selectedProjectID: null,
+      openSection: null,
+    };
+  },
+
+  componentDidMount() {
+    this.context.setAppHeaderVariant('detached');
+    addEventListener('hashChange', this.handleHashChange);
```

Comment:
I'm not entirely sure if `onHashChange` ever triggers. I've tested on localhost+OSX+Chrome53 and localhost+Win7+IE11 by plonking an `alert('I like big bugs and I cannot lie')` in `handleHashChange()`, and it never fires.

If this is true, this may explain why the Ribbons don't work properly in IE11/Edge. IE11 _will_ detect the hash properly, e.g. if I access http://localhost:3735/#focus=projects directly, but it _won't_ detect when the hash has changed to http://localhost:3735/# . I'm still digging through this and will provide a proper report later.